pkg/provisioner/controller.go

```diff
@@ -638,3 +639,38 @@ func (c *obcController) objectBucketForClaimKey(key string) (*v1alpha1.ObjectBuc
 	}
 	return ob, nil
 }
+
+func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucketClaim, class *storagev1.StorageClass) error {
+	ob, err := getObjectBucketFromClaimKey(key, obc, c.libClientset, defaultRetryBaseInterval, defaultRetryTimeout)
+	if err != nil {
+		log.Error(err, "getting ob failed", "key", key)
+		return err
+	}
+	if reflect.DeepEqual(ob.Spec.Endpoint.AdditionalConfigData, obc.Spec.AdditionalConfig) {
+		logD.Info("No updation required")
+		return nil
+	} else {
+		logD.Info("Additional config changed, OB updation required")
+		bucketName, _ := composeBucketName(obc)
+		options := &api.BucketOptions{
+			ReclaimPolicy:     class.ReclaimPolicy,
```

Comment:
The storage class could have been deleted and then recreated with the same name. We don't want to potentially change the retainPolicy. Why are any field being set other than additionalConfigData?

Reply:
I will pass `OBC` instead of options that make more sense, provisioner need to ob info (backend bucket and backend user)

Reply:
What is `additionalState`?

Reply:
I guess I will pass OB than separate parameters, so this API will resemble to Delete() or Grant(). `Additional State` another string map that stores information created by the backend of provisioner like user/owner of the bucket
```console
Spec:
  Additional State:
    Ceph User:  ceph-user-up2Dphkq
```